### PR TITLE
Add timeout to ISession

### PR DESCRIPTION
**Pull request by awharn**
Add timeout to ISession for convenience.
Signed-off-by: Andrew W. Harn <andrew.harn@broadcom.com>

### CODE REVIEW

File: packages/rest/src/session/doc/ISession.ts [hunk 1]
```diff
@@ -114,6 +114,13 @@ export interface ISession {
      */
     rejectUnauthorized?: boolean;
 
+    /**
+     * The amount of time to specify in the X-IBM-Response-Timeout header
```
Comment (dkelosky) [inline packages/rest/src/session/doc/ISession.ts:118]: is this header and implementation more specific to zowe cli & zosmf than imperative?

Comment (awharn) [reply packages/rest/src/session/doc/ISession.ts:118]: The idea to add timeout to the ISession is to make it available for REST clients to be able to extend the base REST client and define a timeout as they see fit. The information provided on that comment in particular was specific to zOSMF due to the testing I was doing, and I forgot to change it to something more generic.

Comment (dkelosky) [reply packages/rest/src/session/doc/ISession.ts:118]: Ok gotcha!

**Comment by zFernand0** [comment]
@awharn, is this supposed to be closed now?

**Comment by awharn** [comment]
That is correct. I will go ahead and close it. This wasn't needed.

**Pull request closed without merging**